atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -358,21 +358,46 @@ def run_image_task(self, image_build_conf):
     def stream_filesystem(self, task_id, filesystem_regex):
         filesystem = self.download_filesystem(task_id, filesystem_regex)
 
-        new_base_image = self.import_base_image(filesystem)
-        self.workflow.builder.set_base_image(new_base_image)
-        defer_removal(self.workflow, new_base_image)
+        new_parent_image = self.import_base_image(filesystem)
+        new_imagename = ImageName.parse(new_parent_image)
 
-        return new_base_image
+        if self.workflow.builder.custom_base_image:
+            self.workflow.builder.set_base_image(new_imagename)
 
-    def run(self):
+        for parent in self.workflow.builder.parent_images.keys():
```

Comment:
So if this a custom base image, we remove any instances of the custom image name in the parent_images as well as the current base_image, but we don't reset custom_base_image?  I think that makes sense, because other plugins need to know that this was a a custom base image, I just want to make sure that is the intent.

Reply:
we are not removing anything from parent_images, but updating value (new image)
for that custom base image, and yes we don't reset custom_base_image or custom_parent_image, as those are flags for other plugins to handle those custom base images